skbeam/core/accumulators/bin_image.py

```diff
@@ -0,0 +1,219 @@
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import numpy as np
+
+
+class BinImage(object):
+    """
+    Class to bin an ndarray onto an axis using a
+    user-specified bin for each element of the array.
+    """
+
+    def __init__(self, bin_values, norm, nbins, min=None, max=None,
+                 weights=None):
+        """
+        Parameters
+        ----------
+        bin_values : np.ndarray
+            For each pixel, this is the value that determines which bin
+            a pixel's intensity will contribute to.  It can be an arbitrary
+            array of floats/ints which will be discretized into nbins equally
```

Comment:
Ah ... i may be understanding (but maybe not).  So you're saying that it would be useful to not only bin in Q, but also delta-Q and that binned_statistic has the ability to do this?